app/services/katello/registration_manager.rb

```diff
@@ -0,0 +1,179 @@
+module Katello
+  class RegistrationManager
+    class << self
+      private :new
+
+      def unregister_host(host, options = {})
+        organization_destroy = options.fetch(:organization_destroy, false)
+        unregistering = options.fetch(:unregistering, false)
+
+        if !organization_destroy && host.subscription_facet.try(:uuid)
+          # if the first operation fails, just raise the error since there's nothing to clean up yet.
+          ::Katello::Resources::Candlepin::Consumer.destroy(host.subscription_facet.uuid)
```

Comment:
I think this should ignore 404s, in case somehow it has already been deleted.

Reply:
I left some duplicated code between the action and the service still around, I'll clean that up which will also handle 410 (consumer deleted) and 404 (consumer never existed 👻  )

Reply:
oh right! 410s are very important too